### ISSUE TITLE
Inconsistent behaviour with Seq '>' and '>=' operators

### ISSUE THREAD

**Issue by chris-rands** [biopython/biopython#1750]
### Setup

I am reporting a problem with Biopython version and operating
system as follows:

```
CPython
Linux-4.17.9-1-ARCH-x86_64-with-arch-Arch-Linux
1.73.dev0
```
The behaviour is different in Python 2 and 3

###  Behaviour

Python 3.6.6:
```
>>> from Bio.Seq import Seq
>>> 'A' > Seq('A')
False
>>> Seq('A') > 'A'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: '>' not supported between instances of 'Seq' and 'str'
```

Python 2.7.15:
```
>>> from Bio.Seq import Seq
>>> 'A' > Seq('A')
False
>>> Seq('A') > 'A'
False
```

It also impacts the `>=` operator in the same way. And it illustrates that the current Python 2 behaviour is wrong:

```
>>> Seq('A') == 'A'
True
>>> 'A' >= Seq('A')
True
>>> Seq('A') >= 'A'
False  # Should be True

```

I'm not exactly sure what changed between Python 2 and 3, but I note that `Seq`, `UnknownSeq`, and `MutableSeq` [implement](https://github.com/biopython/biopython/blob/master/Bio/Seq.py#L224) `__lt__` and `__le__` methods but not `__ge__` or `__gt__` methods (EDIT: in fact, `UnknownSeq` does not implement any of these methods currently).

So the solution might be to implement the `__ge__` or `__gt__` methods (I quickly tested this and it seemed to work). Is this the way forwards?

Also it might be that we want to restrict comparisons for both the `>`/`>=` and the `<`/`<=` case. For example, in Python 2, some pretty funky comparisons are currently allowed:

```
>>> Seq('A') < 1, Seq('A') > 1
(False, True)
>>> Seq('A') < object(), Seq('A') > object()
(False, False)
```

What comparisons should be supported, perhaps only `Seq` to `Seq`,  `Seq` to string, and string to `Seq`? What about for `MutableSeq` and `UnknownSeq`?

(Presumably corresponding unit tests would be needed around [here](https://github.com/biopython/biopython/blob/master/Tests/test_seq.py#L245).)

**Comment by peterjc** [comment]
It looks like an oversight with the way Python 2 and 3 object comparison changed. The design expectation is that ``my_seq`` and ``str(my_seq)`` should behave the same for comparison checks, i.e. act like a plain string.

Do you want to work on this?

**Comment by chris-rands** [comment]
Thanks Peter, I will work on a PR when I find the time

**Comment by chris-rands** [comment]
Fixed in PR #1789
